lib/spack/spack/filesystem_view.py

```diff
@@ -354,6 +393,7 @@ def remove_file(self, src, dest):
         if not os.path.lexists(dest):
             tty.warn("Tried to remove %s which does not exist" % dest)
             return
+        # TODO: Can I remove this?
         if not os.path.islink(dest):
             raise ValueError("%s is not a link tree!" % dest)
```

Comment:
This needs at least to be skipped in the case of relocated files:
```console
$ spack view remove -a ~/tmp/spack-relocate
==> Error: /home/culpo/tmp/spack-relocate/lib/libscalapack.so is not a link tree!
```

Reply:
I think it just needs to always be skipped, as hard links also don't return True for `os.path.islink()`.